src/machine.rs

```diff
@@ -99,14 +108,81 @@ impl Machine {
                 IMAGE,
             ],
         )?;
+
+        let mut container_name = String::new();
+        container_name.push_str("github-self-hosted-runner-");
+        container_name.push_str(&container_id);
+        Self::ssh_exec(
+            &socket_addr,
+            &mut sess,
+            &[
+                "docker",
+                "container",
+                "rename",
+                &container_id,
+                &container_name,
+            ],
+        )?;
+
         info!(
             "[{}] Started a new container: {}",
-            socket_addr, container_id
+            socket_addr, container_name
         );
-
         Ok(())
     }
 
+    /// Returns cache container image is valid
+    /// # Returns
+    ///
+    /// * `Ok(true)` - Container image valid and has not expired
+    /// * `Ok(false)`, `Error` - Container image is no longer valid.
+    fn is_valid_cache_image(
+        socket_addr: &SocketAddr,
+        sess: &mut Session,
+    ) -> Result<bool, Box<dyn Error>> {
+        let dir = Self::ssh_exec(
+            socket_addr,
+            sess,
+            &["echo", "${XDG_CACHE_HOME:-$HOME/.cache}"],
+        )?;
+        let file_name = "gh-actions-scaler";
+        let mut pull_path = String::new();
+        pull_path.push_str(&dir);
+        pull_path.push('/');
+        pull_path.push_str(file_name);
+
+        let now_version = Self::now_cache_version()?;
+
+        let is_exist_file = Self::ssh_exec(socket_addr, sess, &["test", "-f", &pull_path])
+            .map(|_| true)
+            .unwrap_or(false);
+
+        if !is_exist_file {
+            Self::ssh_exec(socket_addr, sess, &["mkdir", "-p", &dir])?;
+            Self::ssh_exec(socket_addr, sess, &["echo", &now_version, ">>", &pull_path])?;
+            return Ok(false);
+        }
+
+        let cached_version = Self::ssh_exec(socket_addr, sess, &["cat", &pull_path])?;
+        let is_valid = cached_version == now_version;
+        if !is_valid {
+            Self::ssh_exec(socket_addr, sess, &["echo", &now_version, ">>", &pull_path])?;
+        }
+        Ok(is_valid)
+    }
+
+    fn now_cache_version() -> Result<String, Box<dyn Error>> {
+        let epoch_seconds = SystemTime::now()
+            .duration_since(UNIX_EPOCH)
+            .map(|duration| duration.as_secs())?;
+
+        //TODO(JopopScript) convert date format yyyyMMdd ex) "20241125"
+        let seconds_a_day = 86400; // 60 * 60 * 24 = 86400
+        let days_since_epoch = epoch_seconds / seconds_a_day;
+        let cache_version = days_since_epoch.to_string();
+        Ok(cache_version)
```

Comment:
- No built-in library exists for date formatting
- third-party library list
  - [chrono:v0.4.38](https://crates.io/crates/chrono)
    - last week commit, recent download 27,827,189
    - datetime formatting unstable feature
  - [jiff:v0.1.14](https://crates.io/crates/jiff)
    - last month commit, recent download 761,352 
  - [time:v0.3.36](https://crates.io/crates/tim)
    - last month commit, recent download 28,005,193
    - document is unkind. can't find date formatting

Reply:
Cache versions are calculated as dates from 1970 to the present and may be less readable.
I don't add libraries because I think this is enough